lib/Core/Completor/NameSearcherCompletor.php

```diff
@@ -52,13 +71,22 @@ protected function createSuggestion(NameSearchResult $result, array $options = [
 
     protected function createSuggestionOptions(NameSearchResult $result, ?TextDocumentUri $sourceUri = null): array
     {
-        return [
-            'short_description' => $result->name()->__toString(),
+        $name = $result->name()->__toString();
+
+        $options = [
+            'short_description' => $name,
             'type' => $this->suggestionType($result),
             'class_import' => $this->classImport($result),
-            'name_import' => $result->name()->__toString(),
-            'priority' => $this->prioritizer->priority($result->uri(), $sourceUri)
+            'name_import' => $name,
+            'priority' => $this->prioritizer->priority($result->uri(), $sourceUri),
         ];
+
+        if ($result->type()->isFunction()) {
```

Comment:
this is a bit of a smell, could we not put this logic in the FunctionCompletor?

Reply:
[This](https://github.com/phpactor/completion/blob/master/lib/Bridge/TolerantParser/WorseReflection/WorseFunctionCompletor.php) FunctionCompletor? Somehow for example `in_array` is found by the `NameSearcherCompletor`.

Reply:
ah that makes sense.

Reply:
we could also add a dedicated `NameSearchResultSnippetFormatter`. But for now I think your change is fine, espeicially as we already have `type()->isClass` in the `classImport` method.

Reply:
Unfortunately the `Formatter` interface requires `format` to return a `string`. I cannot return `null` there. Or I have to create a separate standalone class for that.

Reply:
The `SnippetFormatter` has a `canFormat($object)` method so would look like `if (!$object instanceof NameSearchResult) { return false; } return $object->isFunction()`). Like I say it's up to you, just an idea.

Reply:
I have implemented the `NameSearchResultFunctionSnippetFormatter` as suggested. In addition to that I moved the code to the `NameSearcherCompletor`. In this way the `DoctrineAnnotationCompletor` is not affected by the changes.